pkg/resources/oauth_integration.go

```diff
@@ -0,0 +1,281 @@
+package resources
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/chanzuckerberg/terraform-provider-snowflake/pkg/snowflake"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/pkg/errors"
+)
+
+var oauthIntegrationSchema = map[string]*schema.Schema{
+	"name": {
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+		Description: "Specifies the name of the OAuth integration. This name follows the rules for Object Identifiers. The name should be unique among security integrations in your account.",
+	},
+	"oauth_client": {
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "Specifies the OAuth client type.",
+		ValidateFunc: validation.StringInSlice([]string{
+			"TABLEAU_DESKTOP", "TABLEAU_SERVER", "LOOKER",
+		}, false),
+	},
+	"oauth_issue_refresh_tokens": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Description: "Specifies whether to allow the client to exchange a refresh token for an access token when the current access token has expired.",
+	},
+	"oauth_refresh_token_validity": {
+		Type:        schema.TypeInt,
+		Optional:    true,
+		Description: "Specifies how long refresh tokens should be valid (in seconds). OAUTH_ISSUE_REFRESH_TOKENS must be set to TRUE.",
+	},
+	"oauth_use_secondary_roles": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "Specifies whether default secondary roles set in the user properties are activated by default in the session being opened.",
+		ValidateFunc: validation.StringInSlice([]string{
+			"IMPLICIT", "NONE",
+		}, false),
+	},
+	"blocked_roles_list": {
+		Type:        schema.TypeList,
```

Comment:
```suggestion
		Type:        schema.TypeSet,
```
order shouldn't matter